packages/analytics-adapter/src/awaiter.ts

```diff
@@ -0,0 +1,80 @@
+
+export abstract class Awaiter {
+    public interval: number = 50;
+
+    constructor(public readonly maxWaitTime = 5000) {
+        /*
+        In the extending class, call the following in the constructor
+         this.waitForAnalytics();
+         this.waitForEvolv();
+        */
+    }
+
+    abstract getAnalytics(): any;
+    abstract checkAnalyticsProviders(): void; // Optional check for incorrect configuration
+
+    abstract onAnalyticsFound(analytics: any): void;
```

Comment:
```suggestion
    abstract onAnalyticsFound(analytics: (...args: unknown[]) => void): void;
```

Reply:
Might make sense to create a type now actually:

```typescript
export type AnalyticsHandler<T extends Array<any> = any[]> = (...args: T) => void;
```

Reply:
Will do